packages/env-graph/test/sensitive-decorator.test.ts

```diff
@@ -0,0 +1,81 @@
+import { describe, it, expect } from 'vitest';
+import outdent from 'outdent';
+import { DotEnvFileDataSource, EnvGraph } from '../src';
+
+function sensitiveInferenceTests(
+  tests: Array<{
+    label: string;
+    headers: string;
+    values: string;
+    expected: Record<string, boolean>;
+  }>,
+) {
+  return () => {
+    tests.forEach(({
+      label, headers, values, expected,
+    }) => {
+      it(label, async () => {
+        const g = new EnvGraph();
+        const input = `${headers}\n# ---\n\n${values}`;
+        const testDataSource = new DotEnvFileDataSource('.env.schema', { overrideContents: input });
+        g.addDataSource(testDataSource);
+        await testDataSource.finishInit();
+        await g.finishLoad();
+        for (const key in expected) {
+          const item = g.configSchema[key];
+          expect(item.isSensitive).toBe(expected[key]);
+        }
+      });
+    });
+  };
+}
+
+describe('@defaultSensitive inferFromPrefix', sensitiveInferenceTests([
+  {
+    label: 'key matches prefix is not sensitive',
+    headers: outdent`# @defaultSensitive=inferFromPrefix(PUBLIC_)`,
```

Comment:
this is totally a nit and up to you, but I wonder if it might be cleaner to just have the headers and items together in a single prop.  explicitly writing the divider versus  and extra key and call to outdent :shrug: 
```
  envFile: outdent`
    # @defaultSensitive=inferFromPrefix(PUBLIC_)
    # ---
    PUBLIC_FOO=bar
    SECRET_BAR=baz
 `
 ```